src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -1805,12 +1812,21 @@ class RideConstructionWindow final : public Window
         widgets[WIDX_U_TRACK].type = WindowWidgetType::Empty;
         widgets[WIDX_O_TRACK].type = WindowWidgetType::Empty;
 
-        bool brakesSelected = _selectedTrackType == TrackElemType::Brakes
-            || _currentTrackCurve == (RideConstructionSpecialPieceSelected | TrackElemType::Brakes);
+        bool trackHasSpeedSetting = TrackTypeHasSpeedSetting(_selectedTrackType)
+            || TrackTypeHasSpeedSetting(_currentTrackCurve & ~RideConstructionSpecialPieceSelected);
         bool boosterTrackSelected = _selectedTrackType == TrackElemType::Booster
             || _currentTrackCurve == (RideConstructionSpecialPieceSelected | TrackElemType::Booster);
 
-        if (!brakesSelected && !boosterTrackSelected)
+        // Only necessary because TD6 writes speed and seat rotation to the same bits. Remove for new track design format.
+        bool trackHasSpeedAndSeatRotation = _selectedTrackType == TrackElemType::BlockBrakes
+            || _currentTrackCurve == (RideConstructionSpecialPieceSelected | TrackElemType::BlockBrakes)
+            || _selectedTrackType > TrackElemType::HighestAlias
+            || _currentTrackCurve > (RideConstructionSpecialPieceSelected | TrackElemType::HighestAlias);
```

Comment:
This K-map is the intended use of `trackHasSpeedAndSeatRotation`. The greater than HighestAlias is so all new track elements can have speed and seat rotation because there is no need to scrunch it down for track designs.

ride Has Seat Rotation | track Has Speed Setting | track Has Speed And Seat Rotation | Show BrakeSpeed | Show SeatRotation
--|--|--|--|--
0|0|x|0|0
0|1|x|1|0
1|0|x|0|1
1|1|0|1|0
1|1|1|1|1

Reply:
I cannot say that I understand... why would the Construction window need to concern itself with what TD6 can handle or not?

Reply:
To prevent data loss when exporting to TD6. TD6 can export speed or rotation but not both - this is why seat rotation cannot be set on brakes but can on block brakes.

Reply:
Why is it necessary to check the highest alias?

Reply:
Track elements above HighestAlias cannot be saved in TD6 so there is no requirement for speed and rotation to be exclusive for e.g. diagonal brakes. Block brakes as the exception is detailed more in the speedy block brakes PR.